src/error.rs

```diff
@@ -43,3 +45,67 @@ impl From<std::io::Error> for Error {
         Error::Io(err)
     }
 }
+
+#[macro_export]
+macro_rules! bail {
+    ($e:expr) => {
+        return Err($crate::error::Error::Message($e.to_string()));
+    };
+    ($fmt:expr, $($arg:tt)+) => {
+        return Err($crate::error::Error::Message(format!($fmt, $($arg)+)));
+    };
+}
+
+#[macro_export]
+macro_rules! format_err {
```

Comment:
I'm not sure I'm a huge fan of this macro.  Spelling it out doesn't seem that much work?  But I guess it matches failure's macro so maybe it's fine and I just need to get used to it.

Reply:
it is just a copy of the failures macro, and it makes code a lot more readable